### PR TITLE
feat(civil):  add IsEmpty function to time, date and datetime

### PR DESCRIPTION
**Pull request by Qalifah**

### CODE REVIEW

**Comment by Qalifah** [comment]
> @Qalifah Can you please raise an issue for this feature request for discussion. We generally like to talk things through before accepting PRs, thanks!

I did that https://github.com/googleapis/google-cloud-go/issues/4727

File: civil/civil.go [hunk 1]
```diff
@@ -107,6 +107,11 @@ func (d1 Date) After(d2 Date) bool {
 	return d2.Before(d1)
 }
 
+// IsEmpty reports whether date fields are empty
+func (d Date) IsEmpty() bool {
```
Comment (codyoss) [inline civil/civil.go:111]: I think we should match the API of the time package and use [IsZero](https://pkg.go.dev/time#Time.IsZero).

File: civil/civil.go [hunk 2]
```diff
@@ -107,6 +107,11 @@ func (d1 Date) After(d2 Date) bool {
 	return d2.Before(d1)
 }
 
+// IsEmpty reports whether date fields are empty
```
Comment (codyoss) [inline civil/civil.go:110]: nit: Please end sentence with a period, applies below as well.

**Review by codyoss — CHANGES_REQUESTED** [review]
Just a few suggestions, thanks for the PR.

File: civil/civil.go (hunk 1 above)
Comment (Qalifah) [reply civil/civil.go]: done

File: civil/civil.go (hunk 2 above)
Comment (Qalifah) [reply civil/civil.go]: done

File: civil/civil.go [hunk 3]
```diff
@@ -107,6 +107,11 @@ func (d1 Date) After(d2 Date) bool {
 	return d2.Before(d1)
 }
 
+// IsZero reports whether date fields are empty.
```
Comment (codyoss) [inline civil/civil.go:110]: nit: instead of saying empty for all of these could we say something like ".. are set to their default value"

**Review by codyoss — COMMENTED** [review]
Just one more small thing.

File: civil/civil.go (hunk 3 above)
Comment (Qalifah) [reply civil/civil.go]: okay. Done

**Review by codyoss — APPROVED** [review]
LGTM, thank you for your contribution!

**Pull request merged**